workflow/artifacts/s3/s3.go

```diff
@@ -180,7 +181,23 @@ func (s3Driver *ArtifactDriver) Delete(artifact *wfv1.Artifact) error {
 		if err != nil {
 			return err
 		}
-		return s3cli.Delete(artifact.S3.Bucket, artifact.S3.Key)
+
+		// s3cli.IsDirectory will introduce additional api requests for simple file delete (most scenarios)
```

Comment:
```suggestion
		// check suffix instead of s3cli.IsDirectory as it requires another request for file delete (most scenarios)
```

the comment was a bit unclear to me since it previously mentioned something that the code doesn't use. notate the used technique first, then the unused one for clarity. also simplify a bit of the rest